addons/addon-base-raas/packages/base-raas-services/lib/alb/alb-service.js

```diff
@@ -0,0 +1,314 @@
+    * @param awsAccountId
+    * @param details
+    * @returns {Promise<>}
+    */
+    async saveAlbDetails(awsAccountId, details) {
+        const [deploymentStore] = await this.service(['deploymentStoreService']);
+        return deploymentStore.createOrUpdate({ type: 'account-workspace-details', id: awsAccountId, value: JSON.stringify(details) });
+    }
+
+    /**
+    * Method to get the ALB details.
+    *
+    * @param requestContext
+    * @param projectId
+    * @returns {Promise<>}
+    */
+    async getAlbDetails(requestContext, projectId) {
+        const awsAccountId = await this.findAwsAccountId(requestContext, projectId);
+        return await this.findDeploymentItem({ id: awsAccountId });
+    }
+
+    /**
+    * Method to find the AWS account details for a project.
+    *
+    * @param requestContext
+    * @param projectId
+    * @returns {Promise<>}
+    */
+    async findAwsAccountDetails(requestContext, projectId) {
+        const awsAccountId = await this.findAwsAccountId(requestContext, projectId);
+        const [awsAccountsService] = await this.service(['awsAccountsService']);
+        return await awsAccountsService.mustFind(requestContext, { id: awsAccountId });
+    }
+
+    /**
+    * Method to find the AWS account ID for a project.
+    *
+    * @param requestContext
+    * @param projectId
+    * @returns {Promise<string>}
+    */
+    async findAwsAccountId(requestContext, projectId) {
+        const [
+            indexesService,
+            ProjectService,
+        ] = await this.service([
+            'indexesService',
+            'projectService',
+        ]);
+        const project = await ProjectService.mustFind(requestContext, { id: projectId });
+        const { indexId } = project;
+        // Get the aws account information
+        const { awsAccountId } = await indexesService.mustFind(requestContext, { id: indexId });
+        return awsAccountId;
+    }
+
+    /**
+    * Method to find deployment item from the deployment store table.
+    *
+    * @param id
+    * @returns {Promise<>}
+    */
+    async findDeploymentItem({ id }) {
+        const [deploymentStore] = await this.service(['deploymentStoreService']);
+        return await deploymentStore.find({ type: 'account-workspace-details', id });
+    }
+
+    /**
+    * Method to create listener rule. The method creates rule using the ALB SDK client.
+    * Tags are read form the resolvedVars so the billing will happen properly
+    *
+    * @param prefix
+    * @param requestContext
+    * @param resolvedVars
+    * @param targetGroupArn
+    * @returns {Promise<string>}
+    */
+    async createListenerRule(prefix, requestContext, resolvedVars, targetGroupArn) {
+        const deploymentItem = await this.getAlbDetails(requestContext, resolvedVars.projectId);
+        const albRecord = JSON.parse(deploymentItem.value);
+        const listenerArn = albRecord.listenerArn;
+        const priority = albRecord.albDependentWorkspacesCount + 1;
```

Comment:
Do we need to worry about concurrent requests where `priority` doesn't get updated correctly since the value of `albDependentWorkspacesCount` can be changed in parallel.

Reply:
The concurrent request issues will be resolved once the lock service is implemented. You can expect that in the next PR